### PR TITLE
Implement Shortlimitholdem

### LINKED ISSUE #1: Implement shortlimitholdem game

### PR DESCRIPTION
**Pull request by jake-bickle**
This branch implements Shortlimitholdem game and resolves #1 . 
I've implemented a few test cases to make sure that the few rule changes that differ from regular LimitHoldem are enforced correctly.

Running `python3 -m unittest discover` on `master` and `shortlimitholdem` branches shows that there are no test failures that didn't already exist on the `master` branch.

This branch also implements a bug fix within LimitHoldem, and by extension NoLimitHoldem, where the lowest rank (in these games, any 2 card) was not considered to be apart of a straight hand.

### CODE REVIEW

File: rlcard/games/shortlimitholdem/utils.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        Pick straight cards. Aces count as high and low.
+        Returns:
+            (list): the straight cards
+        '''
+        # Ensures that aces count for low and high cards
+        highest_card = cards[-1]
+        lowest_card = cards[0]
+        if highest_card[1] == 'A':
+            cards.insert(0, highest_card)
+        elif lowest_card[1] == 'A':
```
Comment (jneckar) [inline rlcard/games/shortlimitholdem/utils.py:72]: Why is this necessary?

File: rlcard/games/shortlimitholdem/utils.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            cards.append(lowest_card)
```
Comment (jneckar) [inline rlcard/games/shortlimitholdem/utils.py]: I don't understand why this is necessary, ace should already be in highest card position

File: rlcard/games/shortlimitholdem/utils.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
+from rlcard.games.limitholdem.utils import Hand
+
+class ShortHand(Hand):
+    ''' Employs an alternative ranking system for a short hand game.
+    Said ranks can be found here: https://www.pokernews.com/poker-rules/six-plus-hold-em.htm
+    '''
+    RANK_TO_STRING = {6: "6", 7: "7", 8: "8", 9: "9", 10: "T",
+                      11: "J", 12: "Q", 13: "K", 14: "A"}
+    RANK_LOOKUP = "A6789TJQKA"
```
Comment (jneckar) [inline rlcard/games/shortlimitholdem/utils.py]: Is there a reason for the low A? It seems like the straight evaluator already checks for an ace and inserts it as a low card so I don't think it's necessary

File: rlcard/games/shortlimitholdem/utils.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            self.category = 6
+            #Full house
+            self.best_five = self._get_Fullhouse_cards()
+        elif self._has_three():
+            self.category = 5
+            #Three of a Kind
+            self.best_five = self._get_Three_of_a_kind_cards()
+        elif self._has_straight(self.all_cards):
+            self.category = 4
+            #Straight
```
Comment (jneckar) [inline rlcard/games/shortlimitholdem/utils.py]: Both rule variations exist, but in the more commonly played version, a straight beats 3 of a kind. Let's swap these hand rankings.

File: tests/utils/test_shortlimitholdem_utils.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import unittest
+
+from rlcard.games.shortlimitholdem.utils import ShortHand
+
+class TestShortLimitHoldemUtils(unittest.TestCase):
+    ''' ShortLimitHoldem is not much different from regular LimitHoldem. However, in LimitHoldem,
+    aces only count as a high card, while they count as a low and high card in Short.
+    Because of this, the majority of these tests make sure this works.
```
Comment (jneckar) [inline tests/utils/test_shortlimitholdem_utils.py]: Aces should count as either high or low in any holdem variation, if that wasn't the case for limit holdem then it was a bug in their implementation

**Review by jneckar — CHANGES_REQUESTED** [review]
Great work, there were a few small things I noticed. Check comments in /shortlimitholdem/utils.py

File: rlcard/games/shortlimitholdem/utils.py (hunk 2 above)
Comment (jake-bickle) [reply rlcard/games/shortlimitholdem/utils.py]: That code fixed an issue with the evaluator earlier, but I think I fixed the underlying bug that was causing other issues in the first place so it's indeed redundant. I'll fix it up.

File: rlcard/games/shortlimitholdem/utils.py (hunk 3 above)
Comment (jake-bickle) [reply rlcard/games/shortlimitholdem/utils.py]: The straight evaluator uses RANK_LOOKUP as apart of its logic. It takes a string of ranks and sees if it exists within RANK_LOOKUP. If the low A was not there, the string "A6789" would not exist and incorrectly return false.

File: tests/utils/test_shortlimitholdem_utils.py (hunk 5 above)
Comment (jake-bickle) [reply tests/utils/test_shortlimitholdem_utils.py]: Interesting. I'll fix it up in Limitholdem which should then work for Short as well as No Limit.

**Comment by jake-bickle** [comment]
I removed the redundancies and fixed the underlying bug of aces not counting as highs and lows in straights for LimitHoldem, which also extends to NoLimit and now Short as well. 
I couldn't find any pretrained models for these variants, so I don't think there'll be any side effects to fixing this bug.

**Review by jneckar — APPROVED** [review]
